test/checkpointctl.bats

```diff
@@ -640,3 +640,55 @@ function teardown() {
 	run bash -c "$CHECKPOINTCTL inspect $TEST_TMP_DIR2/test.tar --format=json --sockets | test_socket_src_port"
 	[ "$status" -eq 0 ]
 }
+
+@test "Run checkpointctl list with empty directory" {
+    mkdir "$TEST_TMP_DIR1"/empty
+    checkpointctl list -p "$TEST_TMP_DIR1"/empty/
+    [ "$status" -eq 0 ]
+    [[ ${lines[0]} == *"No checkpoints found"* ]]
+}
+
+@test "Run checkpointctl list with non existing directory" {
+	checkpointctl list -p /does-not-exist
+	[ "$status" -eq 0 ]
+	[[ ${lines[0]} == *"No checkpoints found"* ]]
+}
+
+@test "Run checkpointctl list with empty tar file" {
+	touch "$TEST_TMP_DIR1"/checkpoint-nginx-empty.tar
+	checkpointctl list -p "$TEST_TMP_DIR1"
+	[ "$status" -eq 0 ]
+	[[ "${lines[1]}" == *"Error reading spec.dump file"* ]]
+	[[ "${lines[2]}" == *"Error extracting information"* ]]
+}
+
+@test "Run checkpointctl list with tar file with valid spec.dump and empty config.dump" {
+	touch "$TEST_TMP_DIR1"/config.dump
+	cp data/list_config_spec.dump/spec.dump "$TEST_TMP_DIR1"
+	mkdir "$TEST_TMP_DIR1"/checkpoint
+	( cd "$TEST_TMP_DIR1" && tar cf "$TEST_TMP_DIR2"/checkpoint-config.tar . )
+	checkpointctl list -p "$TEST_TMP_DIR2"
+	[ "$status" -eq 0 ]
+	[[ "${lines[1]}" == *"Error extracting information from $TEST_TMP_DIR2/checkpoint-config.tar: unexpected end of JSON input"* ]]
+}
+
+@test "Run checkpointctl list with tar file with valid config.dump and empty spec.dump" {
+	touch "$TEST_TMP_DIR1"/spec.dump
+	cp data/list_config_spec.dump/config.dump "$TEST_TMP_DIR1"
+	mkdir "$TEST_TMP_DIR1"/checkpoint
+	( cd "$TEST_TMP_DIR1" && tar cf "$TEST_TMP_DIR2"/checkpoint-config.tar . )
+	checkpointctl list -p "$TEST_TMP_DIR2"
+	[ "$status" -eq 0 ]
+	[[ ${lines[1]} == *"Error extracting information from $TEST_TMP_DIR2/checkpoint-config.tar: unexpected end of JSON input" ]]
+}
+
+@test "Run checkpointctl list with tar file with valid config.dump nad spec.dump" {
```

Comment:
Can you create two checkpoint archives to run the list command against. Just to see that we see two checkpoints and not just a single one. You could use `sed` or `jq` to change one of the JSON fields to see two different results.

Reply:
yes sure.